### ISSUE TITLE
Jetpack: Calypso Build Breaks Image Paths From Incorrect `publicPath`

### ISSUE THREAD

**Issue by sdixon194** [Automattic/wp-calypso#56111]
When we updated Jetpack to Webpack 5, it brought along [this change to Calypso Build](https://github.com/Automattic/wp-calypso/pull/54379/files) which sets the publicPath to '/'. This prevented Jetpack's previous default setting of `__webpack_public_path__` from working correctly, which breaks image paths when previewing images in the block editor.

In Jetpack 10.1, [we used a hacky workaround](https://github.com/Automattic/jetpack/pull/20972) to fix this for Jetpack sites, but wanted to get a clearer picture if this is intended behavior from Calypso Build, or if it's possible to provide any option to override that. 

This might feed into a larger discussion about moving Jetpack away from Calypso Build to wp-scripts in the future, but wanted to reach out and get a general idea of what our options are now. 

### Steps to reproduce the behavior
On Jetpack 10.0 or lower (testable using the [Jetpack beta plugin](https://jetpack.com/download-jetpack-beta/) and choosing the 10.0 release):

1. Open a new post
2. In the block selector, hover over something like the Tiled Gallery or Slideshow block to see the block preview pop out.
3. See that the images don't load and GET request errors from the site trying to fetch images from the wrong path.

### Context

p1630700944248800-slack-CBG1CP4EN

#### Browser / OS version

All.

#### Is this specific to the applied theme? Which one?

No. 

#### Does this happen on simple or atomic sites or both?

Both, but fixed for Atomic and self hosted Jetpack sites for now [due to the workaround added here](https://github.com/Automattic/jetpack/pull/20972). 

#### Is there any console output or error text?

Failed GET requests from calling an incorrect file path. 

#### Level of impact (Does it block purchases? Does it affect more than just one site?)

All sites. 

#### Screenshot / Video: If applicable, add screenshots to help explain your problem.

![screen_shot_on_2021-09-03_at_16_43_35](https://user-images.githubusercontent.com/33553323/132563389-ce3ccf34-6183-410c-95ff-69ffb1e53b89.png)

cc @scinos who worked on the update, I believe.

**Comment by scinos** [comment]
Should be relatively easy to fix by deleting the default value for `publicPath`

**Comment by sdixon194** [comment]
Yup, we noticed if that line was removed it worked fine. Wasn't sure if there was any other intended use for it.

**Comment by jordesign** [comment]
This appears to be resolved by https://github.com/Automattic/jetpack/pull/21580
